src/main/java/cc/oakk/lotto/model/WinningLotto.java

```diff
@@ -2,8 +2,25 @@
 
 import java.util.List;
 
-public class WinningLotto extends Lotto {
-    public WinningLotto(List<Integer> numbers) {
-        super(numbers);
+import static cc.oakk.lotto.model.Lotto.NUMBER_COUNT;
+import static cc.oakk.lotto.util.ValidationAdapters.throwIfNull;
+
+public class WinningLotto {
+    private final Lotto lotto;
+    private final LottoNumber bonusNumber;
+
+    public WinningLotto(List<Integer> numbers, int bonusNumber) {
+        this.lotto = new Lotto(numbers);
+        this.bonusNumber = new LottoNumber(bonusNumber);
+    }
+
+    public Rank score(Lotto target) {
+        throwIfNull(target);
+        int matchingCount = (int) lotto.get().stream()
+                .filter(target.get()::contains)
+                .count();
+
+        boolean hasBonus = target.get().stream().anyMatch(v -> v.equals(bonusNumber));
```

Comment:
이정도는 contains()를 사용해도 충분할 것 같습니다:)
```suggestion
        boolean hasBonus = target.get().contains(bonusNumber);
```